### PR TITLE
Add installation instructions for CentOS

### PR DESCRIPTION
**Pull request by mayorova**
Preview [here](https://github.com/3scale/apicast/blob/c65a67cc08d440b40cb236903a02e51b2c37abea/doc/installation.md)

### CODE REVIEW

**Comment by andrewdavidmackenzie** [comment]
Daria, this looks good. But a couple of questions:
- anyone else been able to follow the instructions with success?
- You have WIP in title, and in-progress label.... are you pending anything before final review for potentially merging?

**Comment by mayorova** [comment]
@andrewdavidmackenzie 
Yes, I am pending that somebody else follows the instructions with success :)

Also, 
- was looking for approval from somebody from engineering, which I guess can be considered done
- during the initial installation I had an issue with DNS resolution, which I couldn't reproduce later. There was some recent change related to it, I believe, so it would be good to test the whole process again on the latest revision of `v2`

**Comment by andrewdavidmackenzie** [comment]
You'll need to recruit a CentOS tester @mayorova ...

File: doc/installation.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+```shell
+sudo yum install epel-release
+```
+
+Install LuaRocks:
+```shell
+sudo yum install luarocks
+```
+
+If you are using OAuth authentication method, you will also need to install Redis. This can be done with `sudo yum install redis` command (note that you will need the [EPEL](https://fedoraproject.org/wiki/EPEL) for this).
```
Comment (andrewdavidmackenzie) [inline doc/installation.md]: Is it true that you always need redis?
Only for XC, no?

File: doc/installation.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```shell
+sudo yum install git
+git clone https://github.com/3scale/apicast.git
+cd apicast
+```
+
+You can see all the APIcast releases on the [releases page](https://github.com/3scale/apicast/releases). Use the tags to checkout the vesrion you want to use, for example:
+
+```shell
+git checkout tags/v2.0.0-beta1
```
Comment (andrewdavidmackenzie) [inline doc/installation.md]: it's a moving target I know - but could update this to rc1 now if you wanted...

File: doc/installation.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+Alternatively, you can stay on the latest development branch (`v2`).
+
+Install all the dependencies:
```
Comment (andrewdavidmackenzie) [inline doc/installation.md]: "Install specific Luarocks required by apicast" ??

File: doc/installation.md (hunk 1 above)
Comment (mayorova) [inline doc/installation.md]: As it says, you will need it "if you are using OAuth authentication method" - the only currently supported OAuth flow (authorization code flow, where nginx generates tokens) requires Redis.

File: doc/installation.md (hunk 2 above)
Comment (mayorova) [inline doc/installation.md]: 👍

File: doc/installation.md (hunk 3 above)
Comment (mayorova) [inline doc/installation.md]: Is there a specific version?..
Luarocks installation is covered on line 38 at the moment.

File: doc/installation.md (hunk 2 above)
Comment (mikz) [inline doc/installation.md]: Why not just keep it to `v2` ?

Comment (mikz) [inline doc/installation.md]: I can see it in the next section. I'd keep this out of the installation document. Git basics imo don't belong here.

**Comment by mikz** [comment]
Sometimes the installation documentation is called `INSTALL.md`. 

There are few documents that are in the root like `GETTING_STARTED.md`, `CONTRIBUTING.md`.
Usually the development instructions. Imo it could make sense to follow that and put rest in the `docs` folder .

File: doc/installation.md (hunk 2 above)
Comment (andrewdavidmackenzie) [inline doc/installation.md]: I now see tipo: "vesrion"

**Comment by andrewdavidmackenzie** [comment]
I think best you just propose the actual docs to use and their names Michal - other wise each of the rest of us might have different opinions and different projects we use as references - and we do around and around. The main contribution of the PR is the doc content and I'd avoid getting stuck on exact file naming. You just chose.

**Comment by mikz** [comment]
Yep. Definitely not a blocker for merging. It just might cause some confusion on clear things up like in https://github.com/3scale/apicast/pull/159.

**Comment by mayorova** [comment]
The instructions work on a clean CentOS 7 installation, given that the locales are correctly set. Otherwise, [this bug](https://github.com/3scale/apicast/issues/188) might be hit and prevent the downloaded configuration from being applied correctly.

File: doc/centos-installation.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+If you are using OAuth authentication method, you will also need to install Redis. This can be done with `sudo yum install redis` command (note that you will need the [EPEL](https://fedoraproject.org/wiki/EPEL) for this).
+
+## Install and run APIcast v2
+
+To use the latest APIcast version, you can check out the `v2` branch of the [APIcast GitHub repository](https://github.com/3scale/apicast).
+
+If you want to use a specific version of APIcast, check the APIcast [releases page](https://github.com/3scale/apicast/releases). You can either download the source code from there, or checkout a specific tag with `git`, for example:
+
+```shell
+git checkout tags/v2.0.0-rc1
```
Comment (mikz) [inline doc/centos-installation.md]: I fee like mentioning git is really not needed. Github offers downloads for releases in tar.gz that are checkout from a repo.

**Pull request merged**